sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetFileFormat.scala

```diff
@@ -87,6 +87,22 @@ object GpuParquetFileFormat {
       TrampolineUtil.dataTypeExistsRecursively(field.dataType, _.isInstanceOf[DateType])
     }
 
+    ShimLoader.getSparkShims.getSparkShimVersion.toString() match {
```

Comment:
Main code should never check the specific shim version like this.  It's very fragile and likely to break if there were other versions that had strings like "3.0.1-something".  Put this logic directly in the shim instead rather than making the common code check the shim version before calling the shim.

This comment applies everywhere in the PR where non-shim code is switching on the shim version string.